src/kzg10/data_structures.rs

```diff
@@ -86,6 +87,39 @@ impl<E: PairingEngine> ToBytes for VerifierKey<E> {
     }
 }
 
+/// `PreparedVerifierKey` is the fully prepared version for checking evaluation proofs for a given commitment.
+/// We omit gamma here for simplicity.
+#[derive(Derivative)]
+#[derivative(Default(bound = ""), Clone(bound = ""), Debug(bound = ""))]
+pub struct PreparedVerifierKey<E: PairingEngine> {
+    /// The generator of G1.
+    pub prepared_g: Vec<E::G1Affine>,
```

Comment:
Why not set this to `Vec<E::G1Prepared>`?

Reply:
@weikengchen do you know why?

Reply:
Yes. We should change to E::G1Prepared (without Vec).

The original intention here is to match the constraint world where the prepared key may only support very few bits---but in the end, the constraint world opts to support all the bits, due to the implementation for linear combination.

Note: after changing this, we also need to change the AllocVar for the PreparedVerifierKeyVar in the constraint world, which is in a separate PR.

Reply:
Ok, it is not Vec<E::G1Prepared> because it is just the power series of `g` to help the computation, and it is not used directly as an element for pairing.